src/stream/reader/logfile.rs

```diff
@@ -206,13 +206,16 @@ impl LogFile {
                 Ok(())
             };
 
-        if let Some(task_id) = opts.task {
-            if let Some(task_info) = self.index.get(&task_id) {
-                let instance = task_info.last_instance();
-                if !instance.finished && opts.allow_unfinished {
-                    anyhow::bail!("Stream for task {} is not finished", task_id);
+        if let Some(ref array) = opts.task {
+            for task_id in array.iter() {
+                if let Some(task_info) = self.index.get(&task_id) {
+                    let instance = task_info.last_instance();
+                    if !instance.finished && opts.allow_unfinished {
+                        log::warn!("Stream for task {} is not finished", task_id);
```

Comment:
Look at the code below (for all tasks). First check that all selected tasks are completed, then print them. If at least one of them is not completed (and opts.allow_unfinished is false) then RETURN an error and do not print any of the tasks